core/src/main/java/jenkins/plugins/DetachedPluginsUtil.java

```diff
@@ -0,0 +1,224 @@
+     */
+    private static final Set<String> BREAK_CYCLES;
+
+    static {
+        try (InputStream is = ClassicPluginStrategy.class.getResourceAsStream("/jenkins/split-plugins.txt")) {
+            DETACHED_LIST = ImmutableList.copyOf(configLines(is).map(line -> {
+                String[] pieces = line.split(" ");
+
+                // defaults to Java 1.0 to install unconditionally if unspecified
+                return new DetachedPluginsUtil.DetachedPlugin(pieces[0],
+                                                              pieces[1] + ".*",
+                                                              pieces[2],
+                                                              pieces.length == 4 ? pieces[3] : "1.0");
+            }).collect(Collectors.toList()));
+        } catch (IOException x) {
+            throw new ExceptionInInitializerError(x);
+        }
+        try (InputStream is = ClassicPluginStrategy.class.getResourceAsStream("/jenkins/split-plugin-cycles.txt")) {
+            BREAK_CYCLES = ImmutableSet.copyOf(configLines(is).collect(Collectors.toSet()));
+        } catch (IOException x) {
+            throw new ExceptionInInitializerError(x);
+        }
+    }
+
+    private DetachedPluginsUtil() {
+    }
+
+    /**
+     * Returns all the plugin dependencies that are implicit based on a particular Jenkins version
+     *
+     * @since 2.0
+     */
+    @Nonnull
+    public static List<PluginWrapper.Dependency> getImpliedDependencies(String pluginName, String jenkinsVersion) {
+        List<PluginWrapper.Dependency> out = new ArrayList<>();
+        for (DetachedPlugin detached : getDetachedPlugins()) {
+            // don't fix the dependency for itself, or else we'll have a cycle
+            if (detached.shortName.equals(pluginName)) {
+                continue;
+            }
+            if (BREAK_CYCLES.contains(pluginName + ' ' + detached.shortName)) {
+                LOGGER.log(Level.FINE, "skipping implicit dependency {0} → {1}", new Object[]{pluginName, detached.shortName});
+                continue;
+            }
+            // some earlier versions of maven-hpi-plugin apparently puts "null" as a literal in Hudson-Version. watch out for them.
+            if (jenkinsVersion == null || jenkinsVersion.equals("null") || new VersionNumber(jenkinsVersion).compareTo(detached.splitWhen) <= 0) {
+                out.add(new PluginWrapper.Dependency(detached.shortName + ':' + detached.requiredVersion));
+                LOGGER.log(Level.FINE, "adding implicit dependency {0} → {1} because of {2}",
+                           new Object[]{pluginName, detached.shortName, jenkinsVersion});
+            }
+        }
+        return out;
+    }
+
+    /**
+     * Get the list of all plugins that have ever been {@link DetachedPlugin detached} from Jenkins core, applicable to the current Java runtime.
+     *
+     * @return A {@link List} of {@link DetachedPlugin}s.
```

Comment:
Useless to just repeat the type signature of the method. If you have nothing new to add, delete the tag.